core-api/src/main/java/org/silverpeas/core/contribution/model/WysiwygContent.java

```diff
@@ -0,0 +1,168 @@
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.silverpeas.core.contribution.model;
+
+import org.silverpeas.core.admin.user.model.User;
+import org.silverpeas.core.contribution.service.WysiwygContentRepository;
+import org.silverpeas.core.util.ServiceProvider;
+
+/**
+ * A WYSIWYG content is a rich text (text with styles with graphics and images) that is produced by
+ * an editor in a form closely resembling its appearance when printed or displayed.
+ * <p>
+ * Usually a WYSIWYG content doesn't require to be compiled or transpiled into another form
+ * dedicated to the rendering. The WYSIWYG content is enough by itself as it contains all the
+ * information to be displayed.
+ * @author mmoquillon
+ */
+public class WysiwygContent implements ContributionContent<String> {
+
+  private final LocalizedContribution contribution;
+  private String text;
+  private User author;
+  private boolean modified;
+
+  /**
+   * Gets the WYSIWYG content of the specified contribution.
+   * @param contribution a localized contribution.
+   * @return the WYSIWYG content of the contribution or null if no such content exists.
+   */
+  public static WysiwygContent getContent(final LocalizedContribution contribution) {
+    WysiwygContentRepository repository =
+        ServiceProvider.getService(WysiwygContentRepository.class);
+    return repository.getByContribution(contribution);
+  }
+
+  /**
+   * Deletes all the WYSIWYG contents of the specified contribution. If the given contribution is
+   * an I18n one (a multi-localized contribution) then all of its WYSIWYG contents, one per
+   * localization, will be deleted. In the case of a single one localized contribution, only this
+   * single WYSIWYG content will be deleted. If the contribution has no WYSIWYG contents, nothing is
+   * done.
+   * @param contribution the contribution for which the WYSIWYG contents have to be deleted.
+   */
+  public static void deleteAllContents(final Contribution contribution) {
+    WysiwygContentRepository repository =
+        ServiceProvider.getService(WysiwygContentRepository.class);
+    repository.deleteByContribution(contribution);
+  }
+
+  /**
+   * Constructs a new WYSIWYG content for the specified contribution with the given rich text.
+   * @param contribution the contribution related by the content.
+   * @param richText the data of the content. If null then an empty text is set.
+   */
+  public WysiwygContent(final LocalizedContribution contribution, final String richText) {
+    this.contribution = contribution;
+    this.text = richText == null ? "" : richText;
+  }
+
+  /**
+   * Constructs an empty WYSIWYG content for the specified contribution.
+   * @param contribution the contribution related by the content.
+   */
+  public WysiwygContent(final LocalizedContribution contribution) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1213)